### PR TITLE
Delete ingredient

### PR DESCRIPTION
**Pull request by nicholasblock78**

### CODE REVIEW

File: app/controllers/api/lists_controller.rb [hunk 1]
```diff
@@ -21,17 +21,34 @@ def remove
     end
 
     def save
-      ingredients = current_api_user.kitchen_list.ingredients
-      params[:list][:ingredients].each do |ingredient|
+      params_ingredients = params[:list][:ingredients]
+      orig_ingredients = current_api_user.ingredients
```
Comment (romanjesus) [inline app/controllers/api/lists_controller.rb]: @nicholasblock78 Some time soon let's refactor the var names here to server_ingredients and client_ingredients, that might make it a little clear.

File: app/controllers/api/lists_controller.rb [hunk 2]
```diff
@@ -21,17 +21,34 @@ def remove
...
+      orig_ingredients.each do |orig|
+        if !persisted.include?(orig.id)
+          orig.destroy
+        end
+      end
+
+      #Save new ingredients
+      params_ingredients.each do |ingredient|
         if !ingredient.key?('id')
           Ingredient.create(name: ingredient[:name], kitchen_list_id: 1, category_id: 1)
```
Comment (romanjesus) [inline app/controllers/api/lists_controller.rb]: @nicholasblock78 , can you remove the hard coded kitchen list id here? Switch it to current_api_user.kitchen_list_id 

Otherwise this will only work for our test user

Comment (romanjesus) [reply app/controllers/api/lists_controller.rb]: We added some logic before that whenever a new user is created and they look at their pantry, if they don't have a kitchen list, the server will create one when they look at their pantry so you will have access to it when you try to save to it in this route

File: app/controllers/api/recipes_controller.rb [hunk 3]
```diff
@@ -35,10 +35,11 @@ def favorites
     def ingredients_search
       #Clean up the incoming ingredients so that we can send a clean api request
       #Downcase
-      downcase_ingredients = params["ingredients"].map {|item| item.downcase}
-      #Remove blanks, this validation should be done client side as well
-      ingredients_list = downcase_ingredients.reject { |c| c.empty? }
 
+      # downcase_ingredients = params["ingredients"].map {|item| item.downcase}
```
Comment (romanjesus) [inline app/controllers/api/recipes_controller.rb]: @nicholasblock78 Why did you remove the ingredients validation?

**Review by romanjesus — CHANGES_REQUESTED** [review]
Hey @nicholasblock78 , 

Can you take a look at the comments and make the necessary changes?

**Review by nicholasblock78 — COMMENTED** [review]
should be good to go! @romanjesus
